src/Metadata/Property/Factory/ConstructorPropertyMetadataFactory.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Metadata\Property\Factory;
+
+use ApiPlatform\Core\Api\OperationMethodResolverInterface;
+use ApiPlatform\Core\Exception\PropertyNotFoundException;
+use ApiPlatform\Core\Exception\RuntimeException;
+use ApiPlatform\Core\Metadata\Property\PropertyMetadata;
+
+/**
+ * Set properties available in the constructor as writable.
+ *
+ * @author Maxime Veber <maxime.veber@nekland.fr>
+ */
+final class ConstructorPropertyMetadataFactory implements PropertyMetadataFactoryInterface
+{
+    /**
+     * @var PropertyMetadataFactoryInterface
```

Comment:
Useless (documented by the typehint), to remove.

Reply:
Not all editors / etc uses the constructor typehint for properties, so this looks adequate ?

Reply:
We do that everywhere in the project, and in Symfony as well (we follow Symfony's CS). It's less maintenance and PHPStorm supports this so... 😀

Reply:
yeah, that's why it was more a question, but then I opened a ticket on phpactor. :D